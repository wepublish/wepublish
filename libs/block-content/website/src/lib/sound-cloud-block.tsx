import styled from '@emotion/styled'
import {Block, SoundCloudTrackBlock as SoundCloudTrackBlockType} from '@wepublish/website/api'
import {BuilderSoundCloudTrackBlockProps} from '@wepublish/website/builder'

export const isSoundCloudTrackBlock = (block: Block): block is SoundCloudTrackBlockType =>
  block.__typename === 'SoundCloudTrackBlock'

export const SoundCloudTrackBlockWrapper = styled('div')``

export const SoundCloudTrackBlock = ({trackID, className}: BuilderSoundCloudTrackBlockProps) => (
  <SoundCloudTrackBlockWrapper className={className}></SoundCloudTrackBlockWrapper>
)
