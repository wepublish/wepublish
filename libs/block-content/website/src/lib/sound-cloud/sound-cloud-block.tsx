import styled from '@emotion/styled'
import {Block, SoundCloudTrackBlock as SoundCloudTrackBlockType} from '@wepublish/website/api'
import {BuilderSoundCloudTrackBlockProps} from '@wepublish/website/builder'
import ReactPlayer from 'react-player/soundcloud'

export const isSoundCloudTrackBlock = (block: Block): block is SoundCloudTrackBlockType =>
  block.__typename === 'SoundCloudTrackBlock'

export const SoundCloudTrackBlockWrapper = styled('div')``

export const SoundCloudTrackBlockPlayer = styled(ReactPlayer)`
  width: 100%;
  aspect-ratio: 16/9;
`

export const SoundCloudTrackBlock = ({trackID, className}: BuilderSoundCloudTrackBlockProps) => (
  <SoundCloudTrackBlockWrapper className={className}>
    <SoundCloudTrackBlockPlayer
      width={'auto'}
      height={'auto'}
      url={`https://api.soundcloud.com/tracks/${trackID}`}
      controls={true}
    />
  </SoundCloudTrackBlockWrapper>
)
