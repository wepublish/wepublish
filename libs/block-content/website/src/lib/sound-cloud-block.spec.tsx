import {render} from '@testing-library/react'
import * as stories from './sound-cloud-block.stories'
import {composeStories} from '@storybook/react'

const storiesCmp = composeStories(stories)

describe('SoundCloud Track Block', () => {
  Object.entries(storiesCmp).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      const {asFragment} = render(<Component />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
