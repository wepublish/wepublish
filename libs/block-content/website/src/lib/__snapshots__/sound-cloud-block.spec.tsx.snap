// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundCloud Track Block should render Default 1`] = `
<DocumentFragment>
  <div
    class="css-0 eiisaqe0"
  >
    <div
      class="css-zqjcpl eiisaqe1"
      style="width: auto; height: auto;"
    />
  </div>
</DocumentFragment>
`;

exports[`SoundCloud Track Block should render WithClassName 1`] = `
<DocumentFragment>
  <div
    class="extra-classname css-0 eiisaqe0"
  >
    <div
      class="css-zqjcpl eiisaqe1"
      style="width: auto; height: auto;"
    >
      <iframe
        allow="autoplay"
        frameborder="0"
        src="https://w.soundcloud.com/player/?url=https%3A%2F%2Fapi.soundcloud.com%2Ftracks%2F744469711"
        style="width: 100%; height: 100%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
